File: src/test/resources/mappings/RenameParentChildMethods.java
package mappings;

public class RenameParentChildMethods {

    public static class Parent {
        public void parentMethod() {
        }
    }

    public static class Child extends Parent {
        @Override
        public void parentMethod() {
            super.parentMethod();
        }

        public void childMethod() {
            parentMethod();
        }
    }

    public static void work(Parent parent) {
        parent.parentMethod();
    }

    public static void work(Child child) {
        child.parentMethod();
        child.childMethod();
    }

}

File: src/test/resources/mappings/RenameParentChildMethods.cs
namespace mappings {
    public class RenameParentChildMethods {
        public class Parent {
            public virtual void parentMethod2() {
            }
        }

        public class Child : mappings.RenameParentChildMethods.Parent {
            public override void parentMethod2() {
                base.parentMethod2();
            }

            public virtual void childMethod2() {
                parentMethod2();
            }
        }

        public static void work(mappings.RenameParentChildMethods.Parent parent) {
            parent.parentMethod2();
        }

        public static void work(mappings.RenameParentChildMethods.Child child) {
            child.parentMethod2();
            child.childMethod2();
        }
    }
}